core/src/sumeragi/mod.rs

```diff
@@ -186,7 +187,14 @@ impl SumeragiHandle {
 
     /// Deposit a sumeragi network message.
     pub fn incoming_message(&self, msg: MessagePacket) {
-        if let Err(error) = self.message_sender.try_send(msg) {
+        if matches!(&msg.message, Message::ViewChangeSuggested) {
+            if let Err(error) = self.control_message_sender.try_send(ControlFlowMessage {
+                view_change_proofs: msg.view_change_proofs,
+            }) {
+                self.metrics.dropped_messages.inc();
+                error!(?error, "This peer is faulty. Incoming control messages have to be dropped due to low processing speed.");
+            }
```

Comment:
what happens at this point if a control message is dropped?

Reply:
Consensus tolerates it unless this is happens on more that f peers in the topology 3f + 1

Reply:
github will not allow me comment there but shouldn't [this](https://github.com/hyperledger/iroha/pull/3652/files#diff-900707b74d552f7c23161afeabbf4501209b25c4cb77c61b3aa178baf27a7c00R783) line then be:

```rust
*old_view_change_index = current_view_change_index;
```

Reply:
I think I broke your link by pushing. Can you find it again?

Reply:
[link](https://github.com/hyperledger/iroha/pull/3652/files#diff-900707b74d552f7c23161afeabbf4501209b25c4cb77c61b3aa178baf27a7c00R795)

Reply:
No. Because then rotate_all would be called too few times on the topology.